contracts/reflect/tests/integration.rs

```diff
@@ -18,80 +18,48 @@
-    Api, Instance,
+    testing::{
+        handle, init, mock_env, mock_instance, query, MockApi, MockQuerier, MockStorage,
+        MOCK_CONTRACT_ADDR,
+    },
+    Api, Extern, Instance,
 };
 
-use mock::mock_dependencies;
-use reflect::msg::{CustomMsg, CustomResponse, HandleMsg, InitMsg, OwnerResponse, QueryMsg};
+use reflect::msg::{
+    CustomMsg, CustomQuery, CustomResponse, HandleMsg, InitMsg, OwnerResponse, QueryMsg,
+};
 
 // This line will test the output of cargo wasm
 static WASM: &[u8] = include_bytes!("../target/wasm32-unknown-unknown/release/reflect.wasm");
 // You can uncomment this line instead to test productionified build from cosmwasm-opt
 // static WASM: &[u8] = include_bytes!("../contract.wasm");
 
-mod mock {
-    use reflect::msg::{CustomQuery, CustomResponse};
-
-    use cosmwasm_std::{from_slice, to_binary, Binary, Coin, QueryRequest, StdResult, SystemError};
-    use cosmwasm_vm::{
-        testing::{
-            mock_dependencies as original_mock_dependencies, MockApi, MockQuerier, MockStorage,
-        },
-        Extern, Querier, QuerierResult,
-    };
-
-    #[derive(Clone)]
-    pub struct CustomQuerier {
-        base: MockQuerier,
-    }
-
-    impl CustomQuerier {
-        pub fn new(base: MockQuerier) -> Self {
-            CustomQuerier { base }
-        }
-    }
-
-    impl Querier for CustomQuerier {
-        fn raw_query(&self, bin_request: &[u8]) -> QuerierResult {
-            // parse into our custom query class
-            let request: QueryRequest<CustomQuery> = match from_slice(bin_request) {
-                Ok(v) => v,
-                Err(e) => {
-                    return Ok(Err(SystemError::InvalidRequest {
-                        error: format!("Parsing query request: {}", e),
-                        request: bin_request.into(),
-                    }))
-                }
-            };
-            if let QueryRequest::Custom(custom_query) = &request {
-                Ok(Ok(execute(&custom_query)))
-            } else {
-                self.base.handle_query(&request)
-            }
-        }
-    }
-
-    /// mock_dependencies is a drop-in replacement for cosmwasm_std::testing::mock_dependencies
-    /// this uses our CustomQuerier.
-    pub fn mock_dependencies(
-        canonical_length: usize,
-        contract_balance: &[Coin],
-    ) -> Extern<MockStorage, MockApi, CustomQuerier> {
-        let base = original_mock_dependencies(canonical_length, contract_balance);
-        base.change_querier(CustomQuerier::new)
-    }
-
-    fn execute(query: &CustomQuery) -> StdResult<Binary> {
+/// A drop-in replacement for cosmwasm_vm::testing::mock_dependencies
+/// that supports CustomQuery.
+pub fn mock_dependencies_with_custom_querier(
+    canonical_length: usize,
+    contract_balance: &[Coin],
+) -> Extern<MockStorage, MockApi, MockQuerier<CustomQuery>> {
+    fn execute(query: &CustomQuery) -> MockQuerierCustomHandlerResult {
         let msg = match query {
```

Comment:
It would be cool if we could use `custom_query_execute` here - even if it is not much code in this example, it will be much in others, so nice to ensure this is reusable.

Reply:
But definitely nicer to remove tons of boiler plate

Reply:
Good point. I was not aware of this duplication. Done in faf0298. Feel free to play around with the signatures as you wish.